214 and time tracker model updates

rachelwhaley opened the pull request:
Adding videoconference link field for #214 and adding time tracker model re: #192 (cc @patrickgus)

shiningjustice committed:
edits add button to link to add event

shiningjustice committed:
fixes bug

shiningjustice committed:
fixes bug causing console error

shiningjustice committed:
improves form input styles, addNew display

shiningjustice committed:
adds ability to get userId

shiningjustice committed:
adds event description and TextArea style

shiningjustice committed:
adds remote fields and conditional rendering

shiningjustice committed:
fixes typo

shiningjustice committed:
adds submit function, updates and removes error

shiningjustice committed:
fixes bug for functional radio buttons

shiningjustice committed:
removes error display per bug

shiningjustice committed:
cleans up code

shiningjustice committed:
updates eventType options per Kelsey's wireframe

shiningjustice committed:
adds project name field, cond rendering, and functionality

shiningjustice committed:
adds project name field, cond rendering, and functionality

shiningjustice committed:
conditionally hides remote option if eventType is hacknight

shiningjustice committed:
conditionally hides remote option if eventType is hacknight

shiningjustice committed:
adds recurring event post functionality, refactoring

shiningjustice committed:
adds recurring event post functionality, refactoring

shiningjustice committed:
sends in githubIdentifier field in post

shiningjustice committed:
sends in githubIdentifier field in post

shiningjustice committed:
makes AddEvent comp from AddNew, updates context

shiningjustice committed:
makes AddEvent comp from AddNew, updates context

shiningjustice committed:
fixes bug for default value

shiningjustice committed:
fixes bug for default value

shiningjustice committed:
Improves dropdown placeholder display

shiningjustice committed:
Improves dropdown placeholder display

shiningjustice committed:
adds missing key value on list item from .map

shiningjustice committed:
adds missing key value on list item from .map

shiningjustice committed:
fixes bug for html syntax

shiningjustice committed:
fixes bug for html syntax

shiningjustice committed:
fixes console error, disabled attr

shiningjustice committed:
fixes console error, disabled attr

shiningjustice committed:
improves all radio buttons

shiningjustice committed:
improves all radio buttons

shiningjustice committed:
adds catch chain to promises

shiningjustice committed:
adds catch chain to promises

shiningjustice committed:
fixes bug, enables redirect /events on success

shiningjustice committed:
fixes bug, enables redirect /events on success

sarL3y committed:
Update add event types and recurring event model

sarL3y committed:
Update README

sarL3y committed:
Add weekly  event working

sarL3y committed:
Added conditionals for project data, error boundaries and display with setErrorStatus

sarL3y committed:
Rebase with development

sarL3y committed:
Resolve conflict

shiningjustice committed:
updates addEvent's submitted project fields to match event and recurringevent table models

shiningjustice committed:
adds arrow func for onChange attr of video conference link input

shiningjustice committed:
adds error message if project isn't selected

nickbeaird committed:
Fix unhandled promise warning

patrickgus committed:
add pseudocode for checkIfOnRoster function

patrickgus committed:
add pseudocode for checkIfOnRoster function

patrickgus committed:
add pseudocode for checkIfOnRoster function

patrickgus committed:
add pseudocode for checkIfOnRoster function

patrickgus committed:
add condition to not add member if on roster

patrickgus committed:
add condition to not add member if on roster

patrickgus committed:
add condition to not add member if on roster

patrickgus committed:
add condition to not add member if on roster

nickbeaird committed:
Move files the backend dir for jest setup

OlgaBilogurova committed:
Add /.idea to .gitignore

OlgaBilogurova committed:
Add processData function

OlgaBilogurova committed:
Add functions for processing events and event types

OlgaBilogurova committed:
Create types for chart

OlgaBilogurova committed:
Collect users visited an unique event

OlgaBilogurova committed:
Extract volunteers signedIn and prepare data for 'total volunteers' chart

OlgaBilogurova committed:
Find total volunteer hours and prepare data for 'total hours' chart

OlgaBilogurova committed:
Find volunteer average hours and prepare data for 'total average hours' chart

OlgaBilogurova committed:
Set and display data by default for "All" chart type

OlgaBilogurova committed:
Set data to Donut Charts, fix preloader

patrickgus committed:
remove condition to check if on roster

patrickgus committed:
remove condition to check if on roster

patrickgus committed:
remove condition to check if on roster

patrickgus committed:
remove condition to check if on roster

OlgaBilogurova committed:
Remove previous business logic for calculating stats

OlgaBilogurova committed:
Add UI changes for selecting charts

OlgaBilogurova committed:
Do code refactoring

OlgaBilogurova committed:
Update charts using new colors

OlgaBilogurova committed:
Add color variables for charts

OlgaBilogurova committed:
Fix UI for the dashboard. Remove redundant horizontal & vertical scrollbars,
scrolled background, add proportional vertical margins.

dependabot[bot] committed:
Bump lodash from 4.17.15 to 4.17.19

Bumps [lodash](https://github.com/lodash/lodash) from 4.17.15 to 4.17.19.
- [Release notes](https://github.com/lodash/lodash/releases)
- [Commits](https://github.com/lodash/lodash/compare/4.17.15...4.17.19)

Signed-off-by: dependabot[bot] <support@github.com>

Jonathanko52 committed:
Google Drive Issue completed

Jonathanko52 committed:
Google Drive Issue completed

Jonathanko52 committed:
Can checkin out of meetings. Issue 153

patrickgus committed:
add new route for filtering project team members

patrickgus committed:
add new route for filtering project team members

patrickgus committed:
add new route for filtering project team members

patrickgus committed:
add new route for filtering project team members

patrickgus committed:
create new route that finds user in a project

patrickgus committed:
create new route that finds user in a project

patrickgus committed:
create new route that finds user in a project

patrickgus committed:
create new route that finds user in a project

patrickgus committed:
create function that checks if user is on roster

patrickgus committed:
create function that checks if user is on roster

patrickgus committed:
create function that checks if user is on roster

patrickgus committed:
create function that checks if user is on roster

patrickgus committed:
delete unused route

patrickgus committed:
delete unused route

patrickgus committed:
delete unused route

patrickgus committed:
delete unused route

patrickgus committed:
fix conditions for checkIfOnRoster

patrickgus committed:
fix conditions for checkIfOnRoster

patrickgus committed:
fix conditions for checkIfOnRoster

patrickgus committed:
fix conditions for checkIfOnRoster

alex-anakin committed:
Update environment variables in README

alex-anakin committed:
Merge pull request #230 from hackforla/229-Update-env-variables-in-README

Update environment variables in README

patrickgus committed:
fix route to specify project

patrickgus committed:
fix route to specify project

patrickgus committed:
fix route to specify project

patrickgus committed:
fix route to specify project

patrickgus committed:
fix checkIfOnRoster to return json object

patrickgus committed:
fix checkIfOnRoster to return json object

patrickgus committed:
fix checkIfOnRoster to return json object

patrickgus committed:
fix checkIfOnRoster to return json object

patrickgus committed:
add condition to route for null repsonse

patrickgus committed:
add condition to route for null repsonse

patrickgus committed:
delete unnecessary period

patrickgus committed:
delete unnecessary period

patrickgus committed:
update checkIfOnRoster to set state for isOnRoster

patrickgus committed:
update checkIfOnRoster to set state for isOnRoster

patrickgus committed:
Merge branch 'feature/check-email' of https://github.com/patrickgus/VRMS into feature/check-email

patrickgus committed:
fix parameter for userId

Jonathanko52 committed:
Sending slack messages

rachelwhaley committed:
updates to data model for 214 and time tracker

rachelwhaley committed:
updates to data model for 214 and time tracker

dae-hyun-kim committed:
added autocomplete attribute to input

alex-anakin committed:
Merge pull request #155 from hackforla/feature/add-event

Feature/add event

alex-anakin committed:
Merge pull request #212 from nickbeaird/fix_error_handling

Fix unhandled promise warning

OlgaBilogurova committed:
Merge branch 'development' into 213-recalculate-stats-for-admin-dashboard

# Conflicts:
#	client/src/components/presentational/donutChartContainer.js
#	client/src/pages/AdminDashboard.js

alex-anakin committed:
Merge pull request #246 from OlgaBilogurova/213-recalculate-stats-for-admin-dashboard

213 recalculate stats for admin dashboard

OlgaBilogurova committed:
Merge branch 'development' into 219-fix-broken-UI-admin-dashboard

# Conflicts:
#	client/src/index.scss

alex-anakin committed:
Merge pull request #221 from hackforla/dependabot/npm_and_yarn/lodash-4.17.19

Bump lodash from 4.17.15 to 4.17.19

OlgaBilogurova committed:
Fix height for body in index.scss

alex-anakin committed:
Merge pull request #220 from OlgaBilogurova/219-fix-broken-UI-admin-dashboard

219-Fix broken UI for the dashboard

alex-anakin committed:
Update README to secure environment variables

alex-anakin committed:
Merge pull request #250 from hackforla/secure-env-variables-2

Update README to secure environment variables

patrickgus committed:
delete unused route

patrickgus committed:
add console logs for leader dashboard api calls

Jonathanko52 committed:
console logs/comments cleaned. Posting to test slack channel properly

alex-anakin committed:
Fix finding checkIns by evenId

alex-anakin committed:
Merge pull request #252 from alex-anakin/fix-find-checkIn-by-eventId

Fix finding checkIns by evenId

dae-hyun-kim committed:
Merge branch 'development' into 241-autocomplete-email-input

Jonathanko52 committed:
Beginning Even Check automation

patrickgus committed:
delete console logs and fix bug in addToRoster

patrickgus committed:
Merge branch 'feature/check-email' of https://github.com/patrickgus/VRMS into feature/check-email

alex-anakin commented on models/timeTracker.model.js:
```diff
@@ -0,0 +1,34 @@
+
+const TimeTracker = mongoose.model('TimeTracker', timeTrackerSchema);
+
+module.exports = { timeTracked };
```
please add an empty line at the end

alex-anakin commented on models/timeTracker.model.js:
```diff
@@ -0,0 +1,34 @@
+mongoose.Promise = global.Promise;
+
+const timeTrackerSchema = mongoose.Schema({
+    hours: { type: Number },                // limit to whole numbers or .5 ?
```
it's better to calculate hours in the code using the start time and end time. Let's remove this field.

nickbeaird replied:
++

alex-anakin commented on models/timeTracker.model.js:
```diff
@@ -0,0 +1,34 @@
+    },
+    category: { type: String },             // picklist with 4 options: Development, Design/UX, Product/Project Management, Other
+    notes: { type: String },
+    createdDate: { type: Date, default: Date.now }
```
please add 2 Date fields: startDate, endDate (without default value)

nickbeaird replied:
++

alex-anakin reviewed (changes requested):
Please resolve conflict (apply changes from server to README) and resolve my comments.

OlgaBilogurova committed:
Created _variables.scss file based on the style guide

OlgaBilogurova committed:
Created _main.scss file to store main styles for all app

OlgaBilogurova committed:
Created _elements.scss file to store any generic styles for elements

OlgaBilogurova committed:
Created _keyframes.scss file to store any keyframes or animations for the app

OlgaBilogurova committed:
Created _mixins.scss to store any mixins that will be used over and over

OlgaBilogurova committed:
Added imports all main scss files for implementing a new UI

OlgaBilogurova committed:
Merge branch 'development' into 223-global-styles-to-implement-new-UI

alex-anakin committed:
Merge pull request #243 from dae-hyun-kim/241-autocomplete-email-input

241 Returning Member AutoComplete Email Input

Jonathanko52 committed:
Saving changes to origin

OlgaBilogurova committed:
Merge branch 'development' into 223-global-styles-to-implement-new-UI

patrickgus committed:
delete unused isOnRoster state boolean

patrickgus committed:
added temporary function to fix outdated data

alex-anakin committed:
Merge pull request #222 from Jonathanko52/feature/refactor

Feature/refactor

alex-anakin committed:
Merge pull request #238 from patrickgus/feature/check-email

Feature/check email

OlgaBilogurova committed:
Added font-family 'Open Sans' fonts with formats .woff2, .woff, .ttf

OlgaBilogurova committed:
Added _reset-local.scss to remove built-in browser styling for specific elements

OlgaBilogurova committed:
Added _normalize.scss to provide cross-browser consistency in the default styling of HTML elements

OlgaBilogurova committed:
Created _fonts.scss with @font-face rules allow custom fonts to be loaded on the app

OlgaBilogurova committed:
Created a directive 'variables'
Created separate files to store vars for colors, typography, elements

OlgaBilogurova committed:
Changed import for mixins

OlgaBilogurova committed:
Moved from 'sass' to 'sass-new-ui' directory

OlgaBilogurova committed:
Removed file _variables.scss

OlgaBilogurova committed:
Created '_base.scss' that contains base global styles

OlgaBilogurova committed:
Removed file '_main.scss' because created '_base.scss'

OlgaBilogurova committed:
Changed styles for buttons, added new imports

OlgaBilogurova committed:
Created '_main.scss' to import individual sass files to one

OlgaBilogurova committed:
Included import for '_main.scss'

OlgaBilogurova committed:
Merge branch 'development' into 223-global-styles-to-implement-new-UI

alex-anakin committed:
Merge pull request #253 from OlgaBilogurova/223-global-styles-to-implement-new-UI

Define global styles rules for starting to implement a new UI

clockwerkz committed:
adds empty field check and disable btn styling

nickbeaird committed:
Rebase development onto branch

nickbeaird committed:
Rebase changes on development

nickbeaird committed:
Get backend working with slack

nickbeaird committed:
Get a working client and backend script

nickbeaird committed:
Add backend jest environment

nickbeaird committed:
Get working unit tests with jest

nickbeaird committed:
Do not commit test db or db config file

nickbeaird committed:
Getting a simple api test

nickbeaird committed:
Refactor server and app for testability

Moving the instantiation of the server away from the application itself
allows us to import the express app as a moodule and test it. This keeps
us from actually spinning up a server instance on every import.

nickbeaird committed:
Disable slack app to enable testing

The slack app is starting every time the application is initialized,
which is causing the tests to be unable to close. This work will be
refactored out here soon.

nickbeaird committed:
Add working proxy settings for front and backend

nickbeaird committed:
Get some example test cases

These test cases showcase being able to write unit test and integration
tests using the database, models, and api.

nickbeaird committed:
Fix typo in server file

nickbeaird committed:
Rename test files

nickbeaird committed:
Fix Answer model and add unit test

nickbeaird committed:
Fix Answer model and add unit test

nickbeaird committed:
Add Checkin model unit test

nickbeaird committed:
Add Events unit test

nickbeaird committed:
Get working Project unit test

nickbeaird committed:
Add unit tests for projectTeamMember model

nickbeaird committed:
Return formatting to preserve inline comments

nickbeaird committed:
Add unit tests for Question Model

nickbeaird committed:
Add recurringEvent unit test

nickbeaird committed:
Add user unit test

nickbeaird committed:
Resolve import statements

nickbeaird committed:
Update README for setting up the project

nickbeaird committed:
Add api test for Event creation

nickbeaird committed:
Add documentation for the tests

nickbeaird committed:
Make a workaround to get the tests working

nickbeaird committed:
Reinstate Slack server

nickbeaird committed:
Add in Cypress test framework

nickbeaird committed:
Get a working test

nickbeaird committed:
Move from npm to yarn

The project was having package management version conflicts when using
npm. The suggested solution was to move to yarn, which works.

nickbeaird committed:
Downgrade package versions to work with react reqs

nickbeaird committed:
Add a working frontend unit test

nickbeaird committed:
Get a working Cypress test

nickbeaird committed:
Update the cypress test naming

nickbeaird committed:
Make all of the tests executable

nickbeaird committed:
Remove all hard coded env variables

jafow committed:
Bump nodemon v2.0.4 to get latest security patch;

address an issue in dependency of nodemon. See pr #266 for links to
details.

nickbeaird committed:
Merge pull request #272 from jafow/fix/update-nodemon

Bump nodemon v2.0.4 to get latest security patch;

nickbeaird committed:
Merge remote-tracking branch 'upstream/development' into add_frontend_testing

nickbeaird committed:
Fix merge conflicts with npm vs yarn

nickbeaird committed:
Merge pull request #259 from clockwerkz/bug-fix/disable-check-in

adds empty field check and disable btn styling

nickbeaird committed:
Add changes for better README

clockwerkz committed:
fix for issue #273

nickbeaird committed:
Add changes for review

clockwerkz committed:
refactoring for loop to use for of, removing commented console.logs

clockwerkz committed:
replacing hard coded tokens with env variables

alex-anakin committed:
Merge pull request #270 from nickbeaird/add_frontend_testing

Add testing for application (backend, frontend, end2end)

clockwerkz committed:
Merge branch 'development' of github.com:clockwerkz/VRMS into development
Incorporating new development branch changes

nickbeaird committed:
Fix import error preventing admin from loggin in

nickbeaird committed:
Make the browser tests mor fun

nickbeaird committed:
Merge pull request #275 from nickbeaird/hotfix/admin_login

Fix import error preventing admin from loggin in

nickbeaird committed:
Fixes frontend not loading backend env variable

nickbeaird committed:
Merge pull request #276 from nickbeaird/hotfix/env_files

Hotfix/env files

clockwerkz committed:
Merge branch 'development' of github.com:clockwerkz/VRMS into development

nickbeaird committed:
Redirect React driven router pages

nickbeaird committed:
Raise 404 for all non defined end points

alex-anakin committed:
Merge pull request #277 from nickbeaird/redirect_invalid_urls

Redirect invalid urls

clockwerkz committed:
Merge branch 'development' of https://github.com/clockwerkz/vrms into development

clockwerkz committed:
Merge branch 'development' into fix/recurring-event-exception

alex-anakin committed:
Merge pull request #274 from clockwerkz/fix/recurring-event-exception

fix for issue #273

nickbeaird committed:
Update tests for organization

nickbeaird committed:
Create a user test

nickbeaird committed:
Make ecnryption key more secure

nickbeaird committed:
Get a working authentication verification for user

nickbeaird committed:
Get a working signup end point

nickbeaird committed:
Refactor API validation

nickbeaird committed:
Update tests

nickbeaird committed:
Add middleware

nickbeaird committed:
Get basic Magic Link working

nickbeaird committed:
Cleanup extra fields

alex-anakin committed:
Merge pull request #287 from nickbeaird/magic_link

SPIKE: Magic link

patrickgus committed:
move timeTracker model into backend/models folder

patrickgus committed:
fix requested changes for PR

patrickgus committed:
fix conflicts

patrickgus commented:
@alex-anakin requested changes were made

nickbeaird commented:
It looks like this PR pulled in a ton of prior commits. Is there a way to clean it up, so that it has just your changes? @patrickgus

patrickgus commented:
@nickbeaird Yeah I'm not sure why it pulled so many commits but you can filter the 'files changed' by the individual commits. So mine are the last 3: 4d1e17a, 980a1c5, and 1cee393.

nickbeaird commented:
Hey @patrickgus, I'd prefer it if you can please resubmit the PR/branch if that's the case. I can either review this PR, or the new one then. Also, if you are having any issues getting a branch setup without the commits, then let me know and I can try and help out.

patrickgus commented:
@nickbeaird Should I just delete this PR entirely and create a new one?

nickbeaird commented:
Yeah, I would just make the branch and when you feel good with the new PR, then close this one.

patrickgus commented:
@nickbeaird Just created a new PR #302 that shows only the recent changes. I am closing this one.
